File: Pref360Control/MyTrigger.m
//
//  MyTrigger.m
//  360 Driver
//
//  Created by Pierre TACCHI on 21/01/15.
//

#import "MyTrigger.h"
#import "Pref360StyleKit.h"
#import "ARCBridge.h"

@implementation MyTrigger
@synthesize name = _name;
@synthesize val = _val;

- (void)setName:(NSString *)name {
#if __has_feature(objc_arc)
    _name = name;
#else
    [_name autorelease];
    _name = [name retain];
#endif
    [self setNeedsDisplay:YES];
}

- (void)setVal:(int)value {
    _val = value;
    [self setNeedsDisplay:YES];
}

- (void)drawRect:(NSRect)dirtyRect {
    [super drawRect:dirtyRect];
    
    if (_name == NULL)
        _name = RETAINOBJ(@"");
    [Pref360StyleKit drawTriggerMetterWithIntensity:(_val / 255.0) triggerTitle:_name];
}

@end

File: Pref360Control/MyTrigger.swift
//
//  MyTrigger.swift
//  360 Driver
//
//  Created by C.W. Betts on 5/20/16.
//  Copyright © 2016 GitHub. All rights reserved.
//

import Cocoa

class MyTrigger: NSView {
    var name: String = "" {
        didSet {
            needsDisplay = true
        }
    }
    var val: Int32 = 0 {
        didSet {
            needsDisplay = true
        }
    }

    override func drawRect(dirtyRect: NSRect) {
        super.drawRect(dirtyRect)
        
        Pref360StyleKit.drawTriggerMetterWithIntensity(CGFloat(val) / 255.0, triggerTitle: name)
    }    
}
